src/components/top-bar-equipment-seach-dialog/top-bar-equipment-search-dialog.tsx

```diff
@@ -54,19 +60,40 @@ export const TopBarEquipmentSearchDialog: FunctionComponent<
     const enableSearchDialog = useCallback(() => {
         setIsDialogSearchOpen(true);
     }, [setIsDialogSearchOpen]);
+    const { snackWarning } = useSnackMessage();
 
     const onSelectionChange = useCallback(
         (element: EquipmentInfos) => {
             setIsDialogSearchOpen(false);
             updateSearchTerm('');
             addToLocalStorageSearchEquipmentHistory(studyUuid, element);
-            showVoltageLevelDiagram(element);
+            fetchNetworkElementInfos(
+                studyUuid,
+                currentNode?.id,
+                element.type,
+                EQUIPMENT_INFOS_TYPES.MAP.type,
+                element.id,
+                false
+            )
+                .then(() => {
+                    showVoltageLevelDiagram(element);
```

Comment:
move "addToLocalStorageSearchEquipmentHistory(studyUuid, element);" here
this will add the equipment to the history only if it actually exists